oshi-core/src/main/java/oshi/software/os/windows/WindowsNetworkParams.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Oshi (https://github.com/dblock/oshi)
+ *
+ * Copyright (c) 2010 - 2017 The Oshi Project Team
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ *
+ * Contributors:
+ * https://github.com/dblock/oshi/graphs/contributors
+ */
+package oshi.software.os.windows;
+
+import com.sun.jna.Memory;
+import com.sun.jna.platform.win32.WinDef;
+import com.sun.jna.ptr.IntByReference;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import oshi.jna.platform.windows.IPHlpAPI;
+import oshi.jna.platform.windows.IPHlpAPI.FIXED_INFO;
+import oshi.jna.platform.windows.Kernel32;
+import oshi.software.common.AbstractNetworkParams;
+import oshi.util.platform.windows.WmiUtil;
+
+public class WindowsNetworkParams extends AbstractNetworkParams {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(WindowsNetworkParams.class);
+
+    private static final WmiUtil.ValueType[] GATEWAY_TYPES = {WmiUtil.ValueType.STRING, WmiUtil.ValueType.UINT16};
+    private static final String IPV4_DEFAULT_DEST = "0.0.0.0/0";
+    private static final String IPV6_DEFAULT_DEST = "::/0";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDomainName() {
+        char[] buffer = new char[256];
+        IntByReference bufferSize = new IntByReference(buffer.length);
+        if (!Kernel32.INSTANCE.GetComputerNameEx(Kernel32.ComputerNameDnsDomain, buffer, bufferSize)) {
+            LOG.error("Failed to get dns domain name. Error code: {}", Kernel32.INSTANCE.GetLastError());
+            return "";
+        }
+        return new String(buffer);
```

Comment:
On my test VM, buffer is filled with null characters, so the `String` results in `"\u0000\u0000\u0000\  ... 256 times ... \u0000"`.   You should `return new String(buffer).trim();`.